sieves/tasks/predictive/ner/bridges.py

```diff
@@ -0,0 +1,493 @@
+        for doc_offset in docs_offsets:
+            doc_results = results[doc_offset[0] : doc_offset[1]]
+
+            # Combine all entities from all chunks
+            all_entities: list[Entity] = []
+
+            # Process each chunk for this document
+            for chunk_result in doc_results:
+                if not hasattr(chunk_result, "entities") or not chunk_result.entities:
+                    continue
+
+                # Process entities in this chunk
+                for entity in chunk_result.entities:
+                    all_entities.append(entity)
+
+            # Create a consolidated result for this document
+            yield dspy.Prediction.from_completions({"entities": [all_entities]}, signature=self.prompt_signature)
+
+
+class PydanticBasedNER(NERBridge[pydantic.BaseModel, pydantic.BaseModel, EngineInferenceMode]):
+    @property
+    def _prompt_template(self) -> str | None:
+        return """
+        Your goal is to extract named entities from the text. Only extract entities of the specified types: 
+        {{ entity_types }}.
+
+        {% if examples|length > 0 -%}
+            Examples:
+            ----------
+            {%- for example in examples %}
+                Text: "{{ example.text }}":
+                Entities: [
+                    {%- for entity in example.entities %}
+                    {
+                        "text": "{{ entity.text }}",
+                        
+                        "entity_type": "{{ entity.entity_type }}"
+                    }{%- if not loop.last %}, {% endif %}
+                    {%- endfor %}
+                ]
+            {% endfor -%}
+            ----------
+        {% endif %}
+
+        For each entity:
+        - Extract the exact text of the entity
+        - Include a SHORT context string that contains ONLY the entity and AT MOST 3 words before and 3 words after it. 
+          DO NOT include the entire text as context. DO NOT include words that are not present in the original text 
+          as introductory words (Eg. 'Text:' before context string).
+        - Specify which type of entity it is (must be one of the provided entity types)
+
+        IMPORTANT:
+        - If the same entity appears multiple times in the text, extract each occurrence separately with its own context
+
+        Text: {{ text }}
+        Entity Types: {{ entity_types }}
+        """
+
+    @property
+    def _prompt_signature_description(self) -> str | None:
+        return """
+        A named entity recognition signature that extracts named entities from the provided text. For each entity found:
+        - it extracts the exact text of the entity
+        - it includes a context string that contains ONLY the entity and AT MOST 3 words before and 3 words 
+          after it. DO NOT include the entire text as context.
+        - it specifies the entity type from the provided list of entity types
+        Only extract entities of the specified types.
+        """
```

Comment:
Not used for Pydantic-based engines, set to "".
```suggestion
        return ""
```